src/cryptonote_core/service_node_list.cpp

```diff
@@ -1269,53 +1269,127 @@ namespace service_nodes
     return stream.str();
   }
 
+  static bool is_pulse_block(cryptonote::block const &block)
+  {
+    constexpr cryptonote::pulse_random_value empty_random_value = {};
+
+    bool signatures    = block.signatures.size();
+    bool bitset        = block.pulse.validator_bitset > 0;
+    bool random_value  = !(block.pulse.random_value == empty_random_value);
+    uint8_t hf_version = block.major_version;
+
+    bool result = hf_version >= cryptonote::network_version_16 && (signatures || bitset || random_value);
```

Comment:
Shouldn't we require all of these instead of any of these?

Reply:
Maybe the function can have a better name as it's more of a, does the block have any semblance of Pulse. In the usage code then, if it is, you go and check each Pulse component to make sure it constitutes a valid Pulse block